store/models/job_spec_test.go

```diff
@@ -12,6 +12,37 @@ import (
 	null "gopkg.in/guregu/null.v3"
 )
 
+func TestNewJobFromRequest(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name  string
+		input string
+		want  string
+	}{
+		{"basic",
+			`{"initiators":[{"type":"web"}],"tasks":[{"type":"HttpGet","url":"https://bitstamp.net/api/ticker/"},{"type":"JsonParse","path":["last"]},{"type":"EthBytes32"},{"type":"EthTx"}]}`,
+			"0x57bf5be3447b9a3f8491b6538b01f828bcfcaf2d685ea90375ed4ec2943f4865"},
+		{"downcased types",
+			`{"initiators":[{"type":"web"}],"tasks":[{"type":"httpget","url":"https://bitstamp.net/api/ticker/"},{"type":"jsonparse","path":["last"]},{"type":"ethbytes32"},{"type":"ethtx"}]}`,
+			"0x57bf5be3447b9a3f8491b6538b01f828bcfcaf2d685ea90375ed4ec2943f4865"},
+		{"different cased keys",
+			`{"initiators":[{"type":"web"}],"tasks":[{"TYPE":"httpget","url":"https://bitstamp.net/api/ticker/"},{"type":"jsonparse","path":["last"]},{"type":"ethbytes32"},{"type":"ethtx"}]}`,
+			"0x57bf5be3447b9a3f8491b6538b01f828bcfcaf2d685ea90375ed4ec2943f4865"},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var jsr models.JobSpecRequest
+			assert.Nil(t, json.Unmarshal([]byte(test.input), &jsr))
+
+			js, err := models.NewJobFromRequest(jsr)
+			assert.Nil(t, err)
```

Comment:
Let's lean towards `assert.NoError(t, ...)` or perhaps even `require.NoError(t, ...)` if you want the test to bail earlier.